data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -69,9 +69,8 @@ setOperator
 unaryOperatorExpression
     : primary
     | setInitializer
-    | regexPattern
     | parenthesesExpression
-    | unaryNotOperatorExpression
+    | unaryOperator (primary | unaryOperatorExpression)
```

Comment:
Can this just be `unaryOperator unaryOperatorExpression` because unaryOperatorExpression contains a primary.

Edit: This current definition can lead to an infinite number of `-` and `not`. See the test case examples I proved in ParserTest.java. Is this the right approach going forward as the `-` requires a number where as the `not` requires a boolean? I am thinking we should have an explicit negate and not operator as this would prevent the scenario I outlined.

Reply:
Agreed, ideally the expression parser should not allow `-not 6`. The fix for that would involve reworking the grammar and structure of the parse tree generated.

Data Prepper 1.3.0 will catch the type mismatch exception at statement evaluation time. In future PRs, we can add static type checking (and more).